src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,63 @@
+package duke.tasks;
+
+
+/***
+ * Task class, superclass to Todo, Deadline and Event subclasses
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    //total number of tasks
+    protected static int size = 0;
+
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+        size = size + 1;
+    }
+
+    public String getDescription(){
+        return this.description;
+    }
+
+    public static int getSize(){
+        return size;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /***
+     * Sets isDone to true or false depending on user input
+     * @param action string mark or unmark
+     */
+    public void setAction(String action){
+        if(action.equals("mark")){
+            this.isDone = true;
+            System.out.println("Nice! I've marked this task as done:");
+        } else if(action.equals("unmark")){
+            this.isDone = false;
+            System.out.println("OK, I've marked this task as not done yet:");
+        }
```

Comment:
I like that you prefixed the boolean variable with `is`, so it sounds like a boolean!

Similarly, please add whitespace around Java keywords etc., as per code standards.

```suggestion
        if (action.equals("mark")) {
            this.isDone = true;
            System.out.println("Nice! I've marked this task as done:");
        } else if (action.equals("unmark")) {
            this.isDone = false;
            System.out.println("OK, I've marked this task as not done yet:");
        }
```